src/main/java/com/surrealdb/config/SurrealCrudRepositoryImpl.java

```diff
@@ -0,0 +1,142 @@
+        processSequenceIds(entity);
+        return driver.create(tableName, entity);
+    }
+
+    private void processSequenceIds(T entity) {
+        for (Field field : entity.getClass().getDeclaredFields()) {
+            if (field.isAnnotationPresent(SurrealSequenceId.class)) {
+                field.setAccessible(true);
+                try {
+                    Object currentValue = field.get(entity);
+                    if (currentValue == null) {
+                        String sequenceName = field.getAnnotation(SurrealSequenceId.class).sequenceName();
+                        Long nextIdValue = generateRouteId(sequenceName);
+                        if (field.getType().equals(String.class)) {
+                            field.set(entity, String.valueOf(nextIdValue));
+                        } else if (field.getType().equals(Long.class) || field.getType().equals(long.class)) {
+                            field.set(entity, nextIdValue);
+                        } else if (field.getType().equals(Integer.class) || field.getType().equals(int.class)) {
+                            field.set(entity, nextIdValue.intValue());
+                        } else {
+                            throw new IllegalArgumentException("Unsupported field type for @SurrealSequenceId: " + field.getType());
+                        }
+                    }
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Failed to access or set value for @SurrealSequenceId field", e);
+                }
+            }
+        }
+    }
+
+    private Long generateRouteId(String sequenceName) {
+        String sequenceKey = "sequence:" + tableName + sequenceName;
+        List<QueryResult<SequenceEntity>> response = driver.query(String.format("select seq_value from %s;", sequenceKey), Collections.emptyMap(), SequenceEntity.class);
+        List<SequenceEntity> res = response.stream()
+                .flatMap(qr -> qr.getResult().stream())
+                .toList();
+        if (res.isEmpty()) {
+            return createNewSequence(sequenceKey);
+        }
+        Long currentValue = res.get(0).getSeq_value() + 1;
+        driver.query(String.format("update %s set seq_value = %d;", sequenceKey, currentValue), Collections.emptyMap(), Object.class);
+        return currentValue;
+
+    }
+
+    private Long createNewSequence(String sequenceKey) {
+        driver.query(String.format("create %s SET seq_value = 1", sequenceKey), Collections.emptyMap(), Object.class);
+        return 1L;
+    }
+
+    @Override
+    public void delete(T entity) {
+        driver.delete(getIdFromEntity(entity));
+    }
+
+    @Override
+    public T update(T entity) {
+        return driver.update(getIdFromEntity(entity), entity).get(0);
+    }
+
+    private String getIdFromEntity(T entity) {
+        for (Field field : entity.getClass().getDeclaredFields()) {
+            if (field.isAnnotationPresent(SurrealId.class)) {
```

Comment:
Overall, I'm not opposed to having annotation handling like this. Would be willing to accept annotation handling without spring